packages/proto/src/messages/evm/web3Extension.spec.ts

```diff
@@ -0,0 +1,30 @@
+import { createWeb3Extension } from './web3Extension'
+
+import { ExtensionOptionsWeb3Tx } from '../../proto/ethermint/types/web3'
+
+import { from } from '../../proto/tests/utils'
+import { JSONOptions } from '../../proto/tests/common'
+
+describe('test web3Extension message generation', () => {
+  it('web3Extension', () => {
```

Comment:
are we only testing success cases?

Reply:
Really, there are no failure cases for these methods. The tests ensure that each parameter gets handled as expected, and we don't accidentally do anything stupid, like exclude a field passed in.